tests/parser/test_class2D.py

```diff
@@ -47,53 +56,80 @@ def test_all_keys_are_different(input):
 def test_counter(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    class2d_object._count_all(class_numbers)
+    for i in range(len(class_numbers)):
+        class2d_object._count_all(class_numbers[i])
 
 
 def test_percentage(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    percentage = class2d_object.percent_all_particles_per_class(class_numbers)
-    print("Percent of particles from all data in each class:", percentage)
-    print(
-        "Percent of all particles included in the 20 most populated classes:",
-        sum(x[1] for x in percentage),
-    )
-    assert percentage[0][1] == 18.327067669172934
+    job_numbers = class2d_object.job_number
+    percentage = None
+    for i in range(len(job_numbers)):
+        percentage = class2d_object.percent_all_particles_per_class(class_numbers[i])
+        print("Percent of particles from all data in each class:", percentage)
+        print(
+            "Percent of all particles included in the 20 most populated classes for",
+            job_numbers[i],
+            ":",
+            sum(x[1] for x in percentage),
+            "%",
+        )
+    assert percentage[0][1] == 19.9430499894537
```

Comment:
```suggestion
    assert percentage[0][1] == pytest.approx(19.94305)
```
Testing floating point numbers for equality is always a bit iffy. It's better to compare using [`pytest.approx`](https://docs.pytest.org/en/stable/reference.html#pytest-approx) in those cases, this allows the value to vary by a small amount. It also gives a nice exception message if the value is outside of the allowed range.